### DISCUSSION TITLE
Future roadmap

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1216] [Team Posts]
Apart from the wavefile extension that got a lot of attention due to me, there are a lot of related audio editing features that would be really cool to have and definitely would advance MusE's competitiveness as an audio recording platform. If that is what we want, that is. 

I got a mail from a user asking if there is a roadmap, and we kinda don't have one :) But maybe we should dream up some intentions we have for the short and long term?

Personally my wishlist centers mostly around audio arrangement features (I have to face reality though, I might implement a few but probably not any larger amount...).

Off the top of my head:
- part fade in/out
- part volume
- crossfade between two overlapping parts
- track freeze (which I made some initial work on, in a branch)
- project export for collaboration (export Stems as they call it in at least Pro Tools)

Do you guys have any wishlist/todo list you want to implement or see implemented?

**Comment by kybos** [comment]
Glad you mention it, I've always been convinced a big complex project like MusE needs a roadmap, a vision. People want to know if an app is developing in a direction that's interesting to them, so it make sense for them to keep using it, file in bug reports or even collaborate. 

There is the Project tab in github that can be used quite well for defining a roadmap/wishlist.

That said, I'm more into Midi and in that respect MusE is quite good. What I still miss:
- A working mix cycle recording mode (like in the good old Cubase). It is there but it's quite useless, as it does not play the notes from the previous loop runs, so it can't be used as a (simple) drum machine (which is the main purpose of this mode IMO).
- Audio automation: The visual automation editing is very basic and buggy too (drawing issues). Perhaps the idea is a nonsense, but instead of implementing sophisticated editing specifically for that, it could be simpler to do something like a dynamic input/modulation from existing midi parts (from a specific controller) to control the automation.
- Replace the huge menus (automation, sounds/patches) by more user-friendly UI elements.

**Comment by terminator356** [comment]
> A working mix cycle recording mode (like in the good old Cubase). It is there but it's quite useless, as it does not play the notes from the previous loop runs, so it can't be used as a (simple) drum machine (which is the main purpose of this mode IMO)

Yes that's part of my other branch where I'm fixing all that record mode overdub/replace stuff.
I got soooo off track over the last year.
But the other day I managed to update the branch against current master, and it is functional again. Yay!
Man that was a slog. A lot of hand-merging required :-(
Shows you just how much things change over time. Auto-merging becomes impossible.
And I'm back working on it again with new code.
Please hang in there...

**Comment by spamatica** [comment]
So your main focus kybos, if I may summarize, is fixing the holes that exist in the midi feature list. And I can totally respect that, the MIDI features of MusE are mostly good enough for me but I'm a very basic midi user. 

MusE has always been seen as being very capable for MIDI so actually making it so seems a good idea.

> Audio automation: The visual automation editing is very basic and buggy too (drawing issues). Perhaps the idea is a nonsense, but instead of implementing sophisticated editing specifically for that

> it could be simpler to do something like a dynamic input/modulation from existing midi parts (from a specific controller) to control the automation.

Interesting, could elaborate on how that would appear? 

As I originally added the audio automation in the arranger (it has changed a bit after that) I have always considered it "the way" to work with automation, and for me it works pretty good though I can think of features that would be good, multi-point editing mostly.

What features do you think are missing and what is buggy? It might be that I subconsciously know what isn't working so I don't do problematic operations...

**Comment by kybos** [comment]
Missing features: Yes, multi-point editing, copy-paste, copy automation curves between tracks. Comfort features like fade-in/out, with different slopes, etc.

Perhaps own tracks for automation of individual parameters? We have own track types for everything else...

What I meant with Midi input: Use midi/drum editor to create events to be used as a pattern for automation of audio tracks (one event type like note-on or velocity mapped to one automation parameter). Not sure it's a good idea, I just thought it would be much easier and faster to do that in these advanced editors than with the (currently very simple) automation point editing.

As for the bugs, here some examples. Often the parameter names are drawn outside of the tracks, over each other and over other tracks. The tooltips (which don't seem to use the standard functionality provided by Qt) have many issues as well.

![auto1](https://user-images.githubusercontent.com/10009541/116434547-c91b7c00-a84a-11eb-9072-d7c4ea057a00.png)

![auto2](https://user-images.githubusercontent.com/10009541/116434567-cde03000-a84a-11eb-8d61-84bf3da905a5.png)

(No parameter Amp exists here, BTW, it's only the first word of a multi-word parameter name...)
![auto3](https://user-images.githubusercontent.com/10009541/116434578-d173b700-a84a-11eb-979d-a72389b05af5.png)

![auto4](https://user-images.githubusercontent.com/10009541/116434583-d3d61100-a84a-11eb-89e3-3a2dd5f6aa48.png)

**Comment by terminator356** [comment]
> do something like a dynamic input/modulation from existing midi parts

Agree 100%. This is what I was attempting to do several years ago when I was trying to replace
 the very big audio automation popup menu with something more dialog + table-based.
While I was working on it, it occurred to me to make it an 'uber' controller window where
 all manner of controllers both audio and midi could be routed anywhere to any audio and midi
 controller inputs.
My partial attempt can be seen in control_mapper.cpp/h/ui
It was my first attempt at using new Qt4 'item delegate' features, so I never finished it.
